client/src/Bit.tsx

```diff
@@ -1,20 +1,30 @@
 import * as React from 'react';
 import { Button, Card } from 'react-bootstrap';
+import { connect } from 'react-redux';
 import { Link } from 'react-router-dom';
+import { allActions } from './all_actions';
 import * as styles from './Bit.sass';
 import BitTagPills from './BitTagPills';
-import { USER_DOWNVOTE, USER_UPVOTE } from './reducer';
+import { AppFunctionComponent, NOOP } from './store';
+import { VOTE_DOWN, VOTE_UP } from './store/bits/types';
 import { Bit as BitType } from './types';
 
-export default function Bit(props: any) {
-  const { bit = new BitType(), upvote, downvote } = props;
+type InputProps = {
+  bit: BitType
+};
+
+const getDownvoteButtonStyle = (bit: BitType) => (bit.userVote === VOTE_DOWN ? 'danger' : 'primary');
+const getUpvoteButtonStyle = (bit: BitType) => (bit.userVote === VOTE_UP ? 'success' : 'primary');
+
+const Bit: AppFunctionComponent<InputProps, NOOP> = props => {
+  const { bit, thunkChangeVote } = props;
   const header = (
     <h3>
-      <Button variant={getUpvoteButtonStyle(bit)} className="thumbs-up-button" onClick={() => upvote(bit.postId)}>
+      <Button variant={getUpvoteButtonStyle(bit)} className="thumbs-up-button" onClick={() => thunkChangeVote(bit.postId, VOTE_UP)}>
```

Comment:
Why rename to thunkChangeVote? Do callers need to care that it's a thunk?

Reply:
Yeahh this is one of the reasons why I prefer using classes instead of exporting methods everywhere... Unfortunately the client does need to know that it's a thunk it's calling. Basically every actionCreator and thunk method through the whole app are added to the props object (see client/src/all_actions.ts). If we used the .changeVote() method it would only make this change locally.

Generally how this stuff is laid out is that actioncreators update local state. thunks make API calls and when appropriate update local state via actioncreators. In the future it might be simpler to just delete all of the actioncreator code and directly make them in the thunk methods.

Reply:
Right, but why not just call the thunk changeVote?

Reply:
There is already an action creator called changeVote, and props has all action creators and thunks on it. I prepended "thunk" to all thunk action creators to avoid name collisions.

I'm considering either getting rid of action creators (and simply building them in the thunks) or something. i'm not sure what the best practice is here, it will require some research.